packages/components/src/hooks/useImage.ts

```diff
@@ -0,0 +1,91 @@
+"use client"
+
+import { useState } from "react"
+
+const waitFor = (baseTimeoutMs: number = 500) => {
+  return new Promise((resolve) => setTimeout(resolve, baseTimeoutMs))
+}
+
+const retry = async (
+  promise: () => Promise<string>,
+  baseTimeoutMs: number,
+  maxRetries: number,
+) => {
+  const retryWithBackoff = async (retries: number) => {
+    try {
+      // Make sure we don't wait on the first attempt
+      if (retries > 0) {
+        const timeToWait = baseTimeoutMs * 2 ** retries
+        console.log(`waiting for ${timeToWait}ms...`)
+        await waitFor(timeToWait)
+      }
+      return await promise()
+    } catch (e) {
+      // NOTE: only retry if we didn't reach the limit
+      // otherwise, let the caller handle the error
+      if (retries < maxRetries) {
+        return retryWithBackoff(retries + 1)
+      } else {
+        console.warn("Max retries reached. Bubbling the error up")
+        throw e
+      }
+    }
+  }
+
+  return retryWithBackoff(0)
+}
+
+interface UseImageProps {
+  src: string
+  fallback: string
+  retries?: number
+  baseTimeoutMs?: number
+}
+export const useImage = ({
+  src,
+  fallback,
+  retries = 3,
+  baseTimeoutMs = 500,
+}: UseImageProps) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [isError, setIsError] = useState(false)
+  const [isSuccess, setIsSuccess] = useState(false)
+
+  // NOTE: We require the `useEffect` because `fetch` is a browser API
+  // and it won't exist at compile time for NextJS
```

Comment:
question(non-blocking): is this comment still true? Can't seem to find any usage of `useEffect` here.

Reply:
ah ps, deleted